### PR TITLE
Feature/add optax scheduler

### PR DESCRIPTION
**Pull request by piresramon**
- Rename gin file.
- Add scheduler (for now, using the same arguments used for training gptimbau)
- Set vocab size as 50304 (multiple of 128 for TPU efficiency)

### CODE REVIEW

**Review by rodrigonogueira4 — APPROVED** [review]
Muito obrigado por enviar esse codigo

File: t5x/examples/decoder_only/examples/base_mc4_from_scratch.gin [hunk 1]
```diff
@@ -59,7 +62,13 @@ models.DecoderOnlyModel.inputs_bidirectional_attention = False
...
-  warmup_steps = 10000
+trainer.Trainer:
+  num_microbatches = None
+  learning_rate_fn = @optax.warmup_cosine_decay_schedule()
+
+optax.warmup_cosine_decay_schedule:
+  init_value = 0.0
+  peak_value = 1.2e-5
+  warmup_steps = 13500
+  decay_steps = 135518  %TRAIN_STEPS
```
Comment (ZanezZephyrs) [inline t5x/examples/decoder_only/examples/base_mc4_from_scratch.gin:73]: esse %train_steps era pra ser um comentario, certo? acho que deveria ser # train_steps.

Tentei rodar com essa config e deu erro por causa disso.

**Review by ZanezZephyrs — CHANGES_REQUESTED** [review]

File: t5x/examples/decoder_only/examples/base_mc4_from_scratch.gin (hunk 1 above)
Comment (piresramon) [reply t5x/examples/decoder_only/examples/base_mc4_from_scratch.gin:73]: Verdade, @ZanezZephyrs. Obrigado!

Comment (piresramon) [reply t5x/examples/decoder_only/examples/base_mc4_from_scratch.gin]: Deixei o %TRAIN_STEP na frente (agora comentado) caso optemos por fazer o decay até o final do treino (o GPT-J fez até perto do fim).
